### PR TITLE
dm: add a note about DM to `sql_require_primary_key`

### PR DESCRIPTION
**Pull request by Frank945946**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -825,6 +825,7 @@ mysql> SHOW GLOBAL VARIABLES LIKE 'max_prepared_stmt_count';
 - 这个变量用于控制表是否必须有主键。启用该变量后，如果在没有主键的情况下创建或修改表，将返回错误。
 - 该功能基于 MySQL 8.0 的特性 [`sql_require_primary_key`](https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_sql_require_primary_key)。
 - 强烈推荐在使用 TiCDC 时启用该变量，因为同步数据变更至 MySQL sink 时要求表必须有主键。
+- 如果启用了该变量，且使用了 TiDB DataMigration（DM）来迁移数据，建议在 DM 任务配置文件里的 session 这个参数中设置该 session 变量为 `OFF`，否则会导致 DM 任务创建失败。
```
Comment (hfxsd) [inline system-variables.md:828]:
```suggestion
- 如果启用了该变量，且使用了 TiDB Data Migration (DM) 来迁移数据，建议将 [DM 任务配置文件](/dm/task-configuration-file-full.md#完整配置文件示例)里的 `session` 这个参数设置为 `OFF`，否则会导致 DM 任务创建失败。
```

File: dm/task-configuration-file-full.md [hunk 2]
```diff
@@ -46,6 +46,7 @@ target-database:                # 下游数据库实例配置
     sql_mode: "ANSI_QUOTES,NO_ZERO_IN_DATE,NO_ZERO_DATE" # 从 DM v2.0.0 起，如果配置文件中没有出现该项，DM 会自动从下游 TiDB 中获得适合用于 "sql_mode" 的值。手动配置该项具有更高优先级
     tidb_skip_utf8_check: 1                              # 从 DM v2.0.0 起，如果配置文件中没有出现该项，DM 会自动从下游 TiDB 中获得适合用于 "tidb_skip_utf8_check" 的值。手动配置该项具有更高优先级
     tidb_constraint_check_in_place: 0
+    sql_require_primary_key: OFF   # DM 任务创建期间，会在 TiDB 创建几个元数据库表，这些表，有些表是无主键表。如果开启该参数，这些无主键的表就无法创建出来了。因此需要将该参数设置为 `OFF`。
```
Comment (Frank945946) [inline dm/task-configuration-file-full.md:49]:
```suggestion
    sql_require_primary_key: OFF   # DM 任务创建期间，会在 TiDB 创建几个元数据表，这些表，有些表是无主键表。如果开启该参数，这些无主键的元数据表就无法被创建出来了。因此需要将该参数从 session 级别设置为 `OFF`。
```

**Review by hfxsd — APPROVED** [review]

File: dm/task-configuration-file-full.md [hunk 3]
```diff
@@ -46,6 +46,7 @@ target-database:                # 下游数据库实例配置
     sql_mode: "ANSI_QUOTES,NO_ZERO_IN_DATE,NO_ZERO_DATE" # 从 DM v2.0.0 起，如果配置文件中没有出现该项，DM 会自动从下游 TiDB 中获得适合用于 "sql_mode" 的值。手动配置该项具有更高优先级
     tidb_skip_utf8_check: 1                              # 从 DM v2.0.0 起，如果配置文件中没有出现该项，DM 会自动从下游 TiDB 中获得适合用于 "tidb_skip_utf8_check" 的值。手动配置该项具有更高优先级
     tidb_constraint_check_in_place: 0
+    sql_require_primary_key: OFF   # DM 任务创建期间，会在 TiDB 创建几个元数据表，其中有些表是无主键表。如果开启该参数，这些无主键的元数据表就无法被创建出来。因此需要将该参数从 session 级别设置为 `OFF`。
```
Comment (qiancai) [inline dm/task-configuration-file-full.md:49]:
```suggestion
    sql_require_primary_key: OFF   # 在 session 级别控制表是否必须有主键。DM 任务创建期间，会在 TiDB 创建几个元数据表，其中有些表是无主键表。如果开启该参数，这些无主键的元数据表就无法被创建出来，导致 DM 任务创建失败。因此，需要将该参数设置为 `OFF`。
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

File: system-variables.md [hunk 4]
```diff
@@ -825,6 +825,7 @@ mysql> SHOW GLOBAL VARIABLES LIKE 'max_prepared_stmt_count';
 - 这个变量用于控制表是否必须有主键。启用该变量后，如果在没有主键的情况下创建或修改表，将返回错误。
 - 该功能基于 MySQL 8.0 的特性 [`sql_require_primary_key`](https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_sql_require_primary_key)。
 - 强烈推荐在使用 TiCDC 时启用该变量，因为同步数据变更至 MySQL sink 时要求表必须有主键。
+- 如果启用了该变量，且使用了 TiDB Data Migration (DM) 来迁移数据，建议将 [DM 任务配置文件](/dm/task-configuration-file-full.md#完整配置文件示例)里的 `session` 中的 `sql_require_primary_key` 设置为 `OFF`，否则会导致 DM 任务创建失败。
```
Comment (Frank945946) [inline system-variables.md:828]:
```suggestion
- 如果启用了该变量，且使用了 TiDB Data Migration (DM) 来迁移数据，建议将 [DM 任务配置文件](/dm/task-configuration-file-full.md#完整配置文件示例)里的 `session` 中添加该系统变量 `sql_require_primary_key` 并设置为 `OFF`，否则会导致 DM 任务创建失败。
```

File: system-variables.md [hunk 5]
```diff
@@ -825,6 +825,7 @@ mysql> SHOW GLOBAL VARIABLES LIKE 'max_prepared_stmt_count';
 - 这个变量用于控制表是否必须有主键。启用该变量后，如果在没有主键的情况下创建或修改表，将返回错误。
 - 该功能基于 MySQL 8.0 的特性 [`sql_require_primary_key`](https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_sql_require_primary_key)。
 - 强烈推荐在使用 TiCDC 时启用该变量，因为同步数据变更至 MySQL sink 时要求表必须有主键。
+- 如果启用了该变量，且使用了 TiDB Data Migration (DM) 来迁移数据，建议将 [DM 任务配置文件](/dm/task-configuration-file-full.md#完整配置文件示例)里的 `session` 中添加该系统变量 `sql_require_primary_key` 并设置为 `OFF`，否则会导致 DM 任务创建失败。
```
Comment (qiancai) [inline system-variables.md:828]:
```suggestion
- 如果启用了该变量，且使用了 TiDB Data Migration (DM) 来迁移数据，建议在 [DM 任务配置文件](/dm/task-configuration-file-full.md#完整配置文件示例)里的 `session` 中添加该系统变量 `sql_require_primary_key` 并设置为 `OFF`，否则会导致 DM 任务创建失败。
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Pull request merged**